dolosse/generic_json_consumer.py

```diff
@@ -0,0 +1,142 @@
+"""Reads json data from a kafka topic and graphs it"""
+import json
+import time
+import sys
+import curses
+from matplotlib import pyplot
+import kafka
+
+
+def populate(variables, decode,
+             x_axis_data, prepend=""):
+    """Takes the data from the dict and populates the variables"""
+    for other_key in decode:
+        valid = True
+        name = prepend+other_key
+        try:
+            following = float(decode[other_key])
+        except (ValueError, TypeError):
+            valid = False
+        if isinstance(decode[other_key], dict):
+            populate(variables,
+                     decode[other_key],
+                     x_axis_data,
+                     name+": ")
+        else:
+            if name in variables:
+                if valid:
+                    variables[name].append(following)
+                else:
+                    variables[name].append(
+                        variables[name][-1])
+            else:
+                variables[name] = [0]*(len(x_axis_data)-1)
+                if valid:
+                    variables[name].append(following)
+                else:
+                    variables[name].append(0)
+                    #Remember, the list might still be empty here
+
+
+
+
+
```

Comment:
Should only have 2 lines empty lines between functions.

Reply:
Fixed.